sdk/src/main/java/com/google/cloud/dataflow/sdk/io/BigQueryIO.java

```diff
@@ -539,16 +539,18 @@ private static void dryRunQuery(BigQueryOptions options, String query) {
 
       @Override
       public PCollection<TableRow> apply(PInput input) {
+        BigQueryOptions bqOptions = input.getPipeline().getOptions().as(BigQueryOptions.class);
```

Comment:
Right, BigQuery.Read should validate tempLocation().

Could you move it to validate() above, and also check tempLocation is a gcs path when testServices is not set? Similar as Line 1692 (In BigQuery.Write)

Thanks for pointing out and fixing this bug!